profiles/common/modules/custom/tmgmt_poetry/controller/tmgmt_poetry.controller.job.inc

```diff
@@ -0,0 +1,91 @@
+      $origin_job_items = $origin_job->getItems();
+      $origin_job_item = array_shift($origin_job_items);
+      _tmgmt_poetry_insert_item_status($origin_job_item->tjiid, $origin_job->target_language, "Pending", date('d/m/Y', strtotime($origin_job->settings['delai'])));
+    }
+
+    $origin_job_items = $origin_job->getItems();
+    if ((count($origin_job_items) > 0) && isset($origin_job->settings['languages']) && $translator->plugin == 'poetry') {
+      $origin_job_item = array_shift($origin_job_items);
+      $languages = language_list('language');
+      $requested_languages = $origin_job->settings['languages'];
+      foreach ($requested_languages as $key_requested_language => $requested_language) {
+        if ($key_requested_language != $origin_job->target_language) {
+          if (empty($requested_language)) {
+            // REMOVE A JOB IF IT WAS UNCHECKED.
+            $job_ids_remove = _tmgmt_poetry_obtain_related_translation_jobs(array($key_requested_language), 'SUB_' . $origin_job->tjid)->fetchAll();
+            if (count($job_ids_remove) > 0) {
+              $job_id_remove = array_shift($job_ids_remove);
+              $job_remove = tmgmt_job_load($job_id_remove->tjid);
+              $job_remove->delete();
+            }
+            continue;
+          }
+
+          // CHECK IF JOB IS ALREADY CREATED
+          // AND IN THAT CASE DON'T CREATE IT AGAIN.
+          if (_tmgmt_poetry_obtain_related_translation_jobs(array($requested_language), 'SUB_' . $origin_job->tjid)->rowCount() > 0) {
+            continue;
+          }
+
+          $job = tmgmt_job_create($origin_job->source_language, $requested_language, $origin_job->uid);
+          $job->translator = $origin_job->translator;
+          $job->reference = 'SUB_' . $origin_job->tjid;
+          $job->label = $origin_job->label . ' ' . $languages[$requested_language]->name;
+          $job->save();
+
+          $job_item = tmgmt_job_item_create($origin_job_item->plugin, $origin_job_item->item_type, $origin_job_item->item_id);
+          $job_item->tjid = $job->tjid;
+          $job_item->count_pending = $origin_job_item->count_pending;
+          $job_item->count_translated = $origin_job_item->count_translated;
+          $job_item->count_accepted = $origin_job_item->count_accepted;
+          $job_item->count_reviewed = $origin_job_item->count_reviewed;
+          $job_item->word_count = $origin_job_item->word_count;
+          $job_item->save();
+          _tmgmt_poetry_insert_item_status($job_item->tjiid, $job->target_language, "Pending", date('d/m/Y', strtotime($origin_job->settings['delai'])));
+        }
+      }
+    }
+
+    // IF A POETRY TRANSLATOR HAS ACCEPTED
+    // A TRANSLATION REQUEST WE SHOULD UPDATE
+    // THE RELATED ADDITIONAL SUB TRANSLATION JOBS
+    // WITH A REFERENCE RETURNED BY POETRY.
+    if ($origin_job->state == TMGMT_JOB_STATE_ACTIVE && (strpos($origin_job->reference, 'MAIN_' . $origin_job->tjid . '_POETRY') !== FALSE)) {
```

Comment:
I LIKE THAT THESE COMMENTS
READ LIKE POETRY BUT THEY SHOULD BE
IN HAIKU FORMAT